### PR TITLE
T6910: fix documentation and argspecs are out of alignment

### PR DESCRIPTION
**Pull request by gaige**
## Change Summary

Minor modifications by bringing the `resource_module_builder`-based modules back into line with what is generated, fixing some variances in documentation and inconsistencies in some of the modules and the documentation and
test sources. 

Further, verified and regenerated argspecs based for the newer cli modules.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes)
- [ ] Other (please describe):

## Related Task(s)

[T6910: documentation and argspecs are out of alignment](https://vyos.dev/T6910)

## Proposed changes

Updates to the resource module repo (separate) to bring them into line with the current code.
Added a custom version of the resource_builder in order to ensure we don't remove any
updates for our modules.
Regenerated until we matched all salient updates.

## How to test

```
ansible-test units --docker
ansible-test sanity
ansible-test network-integration --inventory `pwd`/inventory.network
```

Unit tests and sanity tests pass; integration tests are failing in the same ways that the current mainline branch are
failing against 1.4 and 1.5 (but work against 1.3).

## Checklist:

- [x] I have read the [**CONTRIBUTING**](https://github.com/vyos/vyos-1x/blob/current/CONTRIBUTING.md) document
- [x] I have linked this PR to one or more Phabricator Task(s)
- [ ] My commit headlines contain a valid Task id
- [x] My change requires a change to the documentation
- [x] I have updated the documentation accordingly

### CODE REVIEW

**Review by omnom62 — APPROVED** [review]
Looks more cosmetic to me but still good to push

File: docs/vyos.vyos.vyos_ntp_global_module.rst [hunk 1]
```diff
@@ -898,7 +899,7 @@ Common return values are documented `here <https://docs.ansible.com/ansible/late
                             <div>The set of commands pushed to the remote device.</div>
                     <br/>
                         <div style="font-size: smaller"><b>Sample:</b></div>
-                        <div style="font-size: smaller; color: blue; word-wrap: break-word; word-break: break-all;">[&#x27;set service ntp server server1 dynamic&#x27;, &#x27;set service ntp server server1 prefer&#x27;, &#x27;set service ntp server server2 noselect&#x27;, &#x27;set service ntp server server2 preempt&#x27;, &#x27;set service ntp server server_add preempt&#x27;]</div>
```
Comment (gaige) [inline docs/vyos.vyos.vyos_ntp_global_module.rst:901]: Put this back

File: plugins/modules/vyos_ntp_global.py [hunk 2]
```diff
@@ -734,22 +738,17 @@
   returned: when I(state) is C(merged), C(replaced), C(overridden), C(deleted) or C(purged)
   type: list
   sample:
-    - set system ntp server server1 dynamic
```
Comment (gaige) [inline plugins/modules/vyos_ntp_global.py:737]: Revert this change (automated)

File: plugins/module_utils/network/vyos/argspec/bgp_address_family/bgp_address_family.py [hunk 3]
```diff
@@ -34,56 +31,95 @@
...
+                                "as_set": {"type": "bool"},
+                                "summary_only": {"type": "bool"},
+                            },
+                        },
+                        "networks": {
+                            "type": "list",
+                            "elements": "dict",
+                            "options": {
+                                "prefix": {"type": "str"},
+                                "path_limit": {"type": "int"},
```
Comment (omnom62) [inline plugins/module_utils/network/vyos/argspec/bgp_address_family/bgp_address_family.py:58]: My experience was the path_limit was deprecated some time ago - I did not see it in 1.3 nor in newer version. I wanted to re-confirm and remove this option in my work on T6829

**Review by omnom62 — APPROVED** [review]
Thanks - I am not intimately familiar with the code but LGTM!

File: plugins/module_utils/network/vyos/argspec/bgp_address_family/bgp_address_family.py (hunk 3 above)
Comment (gaige) [reply plugins/module_utils/network/vyos/argspec/bgp_address_family/bgp_address_family.py:58]: I'd like to remove it with T6829, to keep the documentation and argspecs in sync with the then-current code. This is intended to bring things to the state of the code at the time of the PR (which is part of why I want to get it in ASAP).

File: docs/vyos.vyos.vyos_firewall_global_module.rst [hunk 4]
```diff
@@ -354,7 +354,7 @@ Parameters
                 <td>
                 </td>
                 <td>
-                        <div>Allows you to specify a brief description for the port group.</div>
+                        <div>A  brief description for the port group.</div>
```
Comment (sever-sever) [inline docs/vyos.vyos.vyos_firewall_global_module.rst:357]: 2 whitespaces after `A`

File: docs/vyos.vyos.vyos_lag_interfaces_module.rst [hunk 5]
```diff
@@ -603,8 +603,7 @@ Examples
     # set interfaces ethernet eth2 bond-group 'bond2'
     # set interfaces ethernet eth3 bond-group 'bond3'
     #
-    - name: Delete LAG attributes of given interfaces (Note This won't delete the interface
-        itself)
+    - name: Delete LAG attributes of given interfaces (Note- This won't delete the interface itself)
```
Comment (sever-sever) [inline docs/vyos.vyos.vyos_lag_interfaces_module.rst:606]: Replace `Note-` to `Note - xxx`, or is this format required?

File: docs/vyos.vyos.vyos_lldp_interfaces_module.rst [hunk 6]
```diff
@@ -397,18 +399,19 @@ Examples
...
     # Module Execution Result
     # -------------------------
     #
-    # before": []
+    # before": {}
     #
     #    "commands": [
     #        "set service lldp interface eth1 location civic-based country-code 'US'",
     #        "set service lldp interface eth1 location civic-based ca-type 0 ca-value 'ENGLISH'",
+    #        "set service lldp interface eth1 location civic-based country-code 'US'",
```
Comment (sever-sever) [inline docs/vyos.vyos.vyos_lldp_interfaces_module.rst:413]: This syntax does not exist `location civic-based`
```
vyos@client-11# set service lldp interface eth1 location 
Possible completions:
 > coordinate-based
                Coordinate based location
   elin         ECS ELIN (Emergency location identifier number)

      
[edit]
vyos@client-11# run show ver

Version:          VyOS 1.3.8
Release train:    equuleus

```

File: plugins/module_utils/network/vyos/argspec/firewall_rules/firewall_rules.py [hunk 7]
```diff
@@ -159,22 +169,23 @@ def __init__(self, **kwargs):
...
-                                },
-                                "jump_target": {
-                                    "type": "str"
+                                    "choices": [
+                                        "match-ipsec",
+                                        "match-none",
+                                        "match-ipsec-in",
+                                        "match-ipsec-out",
+                                        "match-none-in",
+                                        "match-none-out",
```
Comment (sever-sever) [inline plugins/module_utils/network/vyos/argspec/firewall_rules/firewall_rules.py:184]: There are a lot of choices. Are they correct?
```
vyos@r15# set firewall ipv4 input filter rule 100 ipsec 
Possible completions:
   match-ipsec          Inbound IPsec packets
   match-none           Inbound non-IPsec packets


```

File: plugins/module_utils/network/vyos/argspec/lag_interfaces/lag_interfaces.py [hunk 8]
```diff
@@ -41,12 +45,19 @@ def __init__(self, **kwargs):
...
                     },
                     "type": "dict",
                 },
                 "hash_policy": {
-                    "choices": ["layer2", "layer2+3", "layer3+4"],
+                    "choices": [
+                        "layer2",
+                        "layer2+3",
+                        "layer3+4",
+                    ],
```
Comment (sever-sever) [inline plugins/module_utils/network/vyos/argspec/lag_interfaces/lag_interfaces.py:60]: There are more options:
```
vyos@client-11# set interfaces bonding bond0 hash-policy 
Possible completions:
   layer2       use MAC addresses to generate the hash (default)
   layer2+3     combine MAC address and IP address to make hash
   layer3+4     combine IP address and port to make hash
   encap2+3     combine encapsulated MAC address and IP address to make hash
   encap3+4     combine encapsulated IP address and port to make hash
                

```

File: plugins/modules/vyos_firewall_global.py [hunk 9]
```diff
@@ -50,7 +58,8 @@
     type: dict
     suboptions:
       route_redirects:
-        description: -A dictionary of Firewall icmp redirect and source route global
+        description:
+        - A dictionary of Firewall icmp redirect and source route global
```
Comment (sever-sever) [inline plugins/modules/vyos_firewall_global.py:62]: The `ICMP` is an abbreviation

File: plugins/modules/vyos_firewall_global.py [hunk 10]
```diff
@@ -200,7 +211,7 @@
                 required: true
               description:
                 description:
-                - Allows you to specify a brief description for the port group.
+                - A  brief description for the port group.
```
Comment (sever-sever) [inline plugins/modules/vyos_firewall_global.py:214]: two whitespaces

File: docs/vyos.vyos.vyos_firewall_global_module.rst (hunk 4 above)
Comment (gaige) [reply docs/vyos.vyos.vyos_firewall_global_module.rst:357]: These are autogenerated from the `DOCUMENTATION` in the modules, please focus there. It looks like you've also commented on these.

File: plugins/modules/vyos_firewall_global.py [hunk 11]
```diff
@@ -359,6 +374,7 @@
 #        "set firewall config-trap 'enable'",
 #        "set firewall state-policy established action 'accept'",
 #        "set firewall state-policy established log 'enable'",
+#        "set firewall state-policy established log-level 'emerg'",
```
Comment (sever-sever) [inline plugins/modules/vyos_firewall_global.py:377]: Incorrect path (1.3.8)
```
vyos@client-11# set firewall state-policy established log-level

  Configuration path: firewall state-policy established [log-level] is not valid
  Set failed


```

File: docs/vyos.vyos.vyos_lldp_interfaces_module.rst (hunk 6 above)
Comment (gaige) [reply docs/vyos.vyos.vyos_lldp_interfaces_module.rst:413]: `LLDP` module hasn't been updated for 1.3.8, as those changes happen, we'll be updating these items as part of the testing review. This module is consistent with 1.2.

File: plugins/modules/vyos_lldp_interfaces.py [hunk 12]
```diff
@@ -158,18 +167,19 @@
...
 # -------------------------
 #
-# before": []
+# before": {}
 #
 #    "commands": [
 #        "set service lldp interface eth1 location civic-based country-code 'US'",
 #        "set service lldp interface eth1 location civic-based ca-type 0 ca-value 'ENGLISH'",
+#        "set service lldp interface eth1 location civic-based country-code 'US'",
+#        "set service lldp interface eth1 location civic-based ca-type 0 ca-value 'ENGLISH'",
```
Comment (sever-sever) [inline plugins/modules/vyos_lldp_interfaces.py:182]: Invalid path for 1.3.8
```
vyos@client-11# set service lldp interface eth1 location civic-based 

  Configuration path: service lldp interface eth1 location [civic-based] is not valid

```

File: plugins/module_utils/network/vyos/argspec/firewall_rules/firewall_rules.py (hunk 7 above)
Comment (gaige) [reply plugins/module_utils/network/vyos/argspec/firewall_rules/firewall_rules.py:184]: These have to support all versions that we support, which for this module is 1.3-1.5. 
```
vyos@vv15.dc.cluetrust.com# set firewall ipv4 input filter rule 100 ipsec 
Possible completions:
   match-ipsec-in       Inbound traffic that was IPsec encapsulated
   match-none-in        Inbound traffic that was not IPsec encapsulated
vyos@vv15.dc.cluetrust.com# set firewall ipv4 output  filter rule 100 ipsec 
Possible completions:
   match-ipsec-out      Outbound traffic to be IPsec encapsulated
   match-none-out       Outbound traffic that will not be IPsec encapsulated
vyos@vv15.dc.cluetrust.com# set firewall ipv4 forward filter rule 100 ipsec 
Possible completions:
   match-ipsec-in       Inbound traffic that was IPsec encapsulated
   match-ipsec-out      Outbound traffic to be IPsec encapsulated
   match-none-in        Inbound traffic that was not IPsec encapsulated
   match-none-out       Outbound traffic that will not be IPsec encapsulated
vyos@vv15.dc.cluetrust.com:~$ show version 
Version:          VyOS 1.5-rolling-202411070006
Release train:    current
```

File: plugins/module_utils/network/vyos/argspec/lag_interfaces/lag_interfaces.py (hunk 8 above)
Comment (gaige) [reply plugins/module_utils/network/vyos/argspec/lag_interfaces/lag_interfaces.py:60]: Argspec are auto-generated from the `DOCUMENTATION` in the modules themselves, please focus on those unless they're inconsistent.

For LAG that module is still only 1.2, so those options will show up when they're supported. The goal of this PR is to get the documentation differences from the current code resolved.

File: plugins/modules/vyos_firewall_global.py (hunk 11 above)
Comment (gaige) [reply plugins/modules/vyos_firewall_global.py:377]: Good catch, this module was updated, docs were not.

File: plugins/modules/vyos_lldp_interfaces.py (hunk 12 above)
Comment (gaige) [reply plugins/modules/vyos_lldp_interfaces.py:182]: lldp hasn't been updated to 1.3.8 from 1.2 yet. That will come when we update the code for 1.3+

**Review by sever-sever — APPROVED** [review]
